substrate-airgapped/src/tx/generic_call.rs

```diff
@@ -9,21 +9,21 @@ pub struct CallIndex {
 }
 
 impl CallIndex {
-	/// Get `module`, the index of the module the call is in
+	/// Get `module`, the index of the module the call is in.
 	pub fn module_index(&self) -> u8 {
 		self.module_index
 	}
-	/// Get `call_index`, the index of the call in the module
+	/// Get `call_index`, the index of the call in the module.
 	pub fn call_index(&self) -> u8 {
 		self.call_index
 	}
 	/// Create `CallIndex`
 	pub fn new(module_index: u8, call_index: u8) -> CallIndex {
 		CallIndex { module_index, call_index }
 	}
-	/// Create a vec representing the call index
-	pub fn to_vec(&self) -> Vec<u8> {
-		vec![self.module_index, self.call_index]
+	/// Create a `[u8; 2]` representing the call index.
+	pub fn to_bytes(&self) -> [u8; 2] {
```

Comment:
Completely random thought: wonder if it's possible to make the compiler take a `CallIndex` and use it as a `[u8; 2]` directly, without any code at all. In memory I don't think there should be any difference at all between the two; maybe there's some `#[repr(…)]` trick to make it interchangeable? Would require `unsafe` and `transmute` I think so definitely not a good idea, but if it worked it'd be cool.